probe-rs/src/architecture/arm/memory/adi_v5_memory_interface.rs

```diff
@@ -166,8 +181,19 @@ where
         // Offset of byte in word (little endian)
         let bit_offset = (address - aligned.start) * 8;
 
-        // Read 32-bit word and extract the correct byte
-        let result = ((self.read32(aligned.start)? >> bit_offset) & 0xFF) as u8;
+        let result = if self.only_32bit_data_size {
+            // Read 32-bit word and extract the correct byte
+            ((self.read32(aligned.start)? >> bit_offset) & 0xFF) as u8
+        } else {
+            let csw = self.build_csw_register(DataSize::U8);
+            let tar = TAR { address };
+            self.write_ap_register(csw)?;
+            self.write_ap_register(tar)?;
+            let result = self.read_ap_register(DRW::default())?;
+
+            // Extract the correct byte
+            ((result.data >> bit_offset) & 0xFF) as u8
```

Comment:
I still think this line is not needed anymore, is it? At least not for 8/16 bit ready, it is for 32bits.

Reply:
It's still needed for 8/16 bit reads.

Reply:
How so, oO does it still return 32bits in full?

Reply:
`read_ap_register` reads 32 bits and I need to extract 8 bits from this result. The offset of these 8 bits depends on the least significant bits of the address.

Reply:
Could you please add a reference to the documentation for this? Just document name and chapter, to make it easier in the future

Reply:
Oh, I figured this out with my logic analyzer, I need time to find this in docs